rotkehlchen/chain/base/modules/runmoney/decoder.py

```diff
@@ -0,0 +1,174 @@
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import bytes_to_address
+
+from .constants import (
+    CLAIM_TOPIC,
+    CPT_RUNMONEY,
+    JOINED_TOPIC,
+    RECORD_ACTIVITY_TOPIC,
+    RUNMONEY_CONTRACT_ADDRESS,
+    RUNMONEY_MEMBERSHIP_NFT_COLLECTION_IDENTIFIER,
+    UNSTAKE_TOPIC,
+)
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class RunmoneyDecoder(DecoderInterface):
+    def _decode_join_event(self, context: DecoderContext) -> DecodingOutput:
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.asset == self.evm_inquirer.native_token
+            ):
+                event.counterparty = CPT_RUNMONEY
+                event.event_subtype = HistoryEventSubType.FEE
+                event.notes = f'Pay {event.amount} ETH membership fee to join Runmoney club'
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    tokenid_belongs_to_collection(event.asset.identifier, RUNMONEY_MEMBERSHIP_NFT_COLLECTION_IDENTIFIER)  # noqa: E501
+            ):
+                event.counterparty = CPT_RUNMONEY
+                event.notes = f'Receive Runmoney membership NFT with id {tokenid_to_collectible_id(event.asset.identifier)} for joining the club'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    @staticmethod
+    def _decode_stake_unstake_event(
+            context: DecoderContext,
+            from_event_type: Literal[HistoryEventType.SPEND, HistoryEventType.RECEIVE],
+            to_event_type: Literal[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL],
+            to_event_subtype: Literal[HistoryEventSubType.DEPOSIT_ASSET, HistoryEventSubType.REMOVE_ASSET],  # noqa: E501
+            action: Literal['deposit', 'withdraw'],
+            preposition: Literal['into', 'from'],
+    ) -> DecodingOutput:
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=6,  # usdc token has decimal=6
+        )
+        for event in context.decoded_events:
+            if (
+                    event.event_type == from_event_type and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount
+            ):
+                event.counterparty = CPT_RUNMONEY
+                event.event_type = to_event_type
+                event.event_subtype = to_event_subtype
+                event.notes = f'{action.capitalize()} {amount} USDC {preposition} Runmoney'
```

Comment:
a bit of a nitpick but the preposition doesn't need to be an argument. YOu can do an if just checking the type